src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -799,6 +816,24 @@ private static EntitySourceType GetEntitySourceType(string entityName, Entity en
             }
         }
 
+        /// <summary>
+        /// Helper method to create a linking entity and a database object for the given linking object (which relates the source and target with an M:N relationship).
+        /// The created linking entity and its corresponding database object definition is later used during GraphQL schema generation
+        /// to enable nested mutations.
+        /// </summary>
+        /// <param name="entityName">Source entity name.</param>
+        /// <param name="targetEntityName">Target entity name.</param>
+        /// <param name="linkingObject">Linking object</param>
+        /// <param name="sourceObjects">Dictionary storing a collection of database objects which have been created.</param>
+        protected virtual void PopulateMetadataForLinkingObject(
+            string entityName,
+            string targetEntityName,
+            string linkingObject,
+            Dictionary<string, DatabaseObject> sourceObjects)
+        {
+            return;
```

Comment:
Similar kind of reasoning here:

https://github.com/Azure/data-api-builder/pull/1902#discussion_r1474017853

The implementation will be same across all the dbs for this method. Just that we, for now, are only supporting nested insertion for MsSql. When we will support it for the other dbs, we can just move the method implementation to the base method. We can throw a not implemented exception, but then this would cascade into other conditionals being put on the database type in the code which is present in the parent class i.e. SqlMetadataProvider, so that we make sure that this method is only called for certain sub-classes (database flavors) and not for others.